athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -172,21 +178,25 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
     {
         logger.debug("doListTables: enter - " + request);
 
-        List<TableName> indices = new ArrayList<>();
-
         try {
             String endpoint = getDomainEndpoint(request.getSchemaName());
             AwsRestHighLevelClient client = clientFactory.getOrCreateClient(endpoint);
             try {
```

Comment:
Why do you have the nested trys here?

If you're worried about logging, just log before something critical is done and then just let the exception bubble up as it does without catching it (other than checked exceptions that have to be rethrown).

This way, if you think about it, when the exception happens and it errors out, we have to look at the logs anyway, and then we'll just see in the logs the last log of what we were trying to do before the exception happened.

Additionally, even without that log message, we know exactly where to look because the exception will contain the appropriate stack trace already.